### PR TITLE
virtio_console_guest:fix wrong way to store hvc with fixed kernel

### PR DESCRIPTION
**Pull request by nanliu-r**
For 'is_console': yes, vtermno will start from '1..', which was
previously starting from '0'. 

ID:1770165

Signed-off-by: Nana Liu <nanliu@redhat.com>

### CODE REVIEW

**Comment by nanliu-r** [comment]
Test results:
 (1/1) Host_RHEL.m8.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.virtio_console.spread_linear.unix_socket.virtserialport_test.restart.stressed: PASS (145.93 s)

Host_RHEL.m7.u8.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.8.x86_64.io-github-autotest-qemu.virtio_console.spread_linear.unix_socket.virtserialport_test.restart.stressed: PASS (227.75 s)

File: shared/scripts/virtio_console_guest.py [hunk 1]
```diff
@@ -273,7 +291,10 @@ def _get_port_status(self, in_files=None):
                         port[m.group(1)] = m.group(2)
 
                     if port['is_console'] == "yes":
-                        port["path"] = "/dev/hvc%s" % (port["console_vtermno"])
+                        if fix_kernel_version:
```
Comment (xutian) [inline shared/scripts/virtio_console_guest.py]: Let's make code more readable,  I guess you want to change base port number by kernel.  let's give your some example code in below:

```
import platform
from distutils.version import LooseVersion
current_version = LooseVersion(platform.uname()[2])
fixed_version = = LooseVersion("4.18.0-97")
port_num = int(port["console_vtermno"])
path  = "/dev/hvc%d" % ( port_num if  current_version >= fixed_version  else port_num -1 )
```

Comment (nanliu-r) [reply shared/scripts/virtio_console_guest.py]: Thanks and update, please help to review this.

**Comment by nanliu-r** [comment]
> Let's make code more readable.
thanks.

**Comment by nanliu-r** [comment]
Test results:
 (1/2) Host_RHEL.m8.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.virtio_console.spread_linear.unix_socket.virtserialport_test.restart.stressed: PASS (127.40 s)
 (2/2) Host_RHEL.m8.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.virtio_console.spread_2.unix_socket.virtserialport_test.restart.stressed: PASS (134.98 s)

 (1/2) Host_RHEL.m7.u8.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.8.x86_64.io-github-autotest-qemu.virtio_console.spread_linear.unix_socket.virtserialport_test.restart.stressed: PASS (244.70 s)
 (2/2) Host_RHEL.m7.u8.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.8.x86_64.io-github-autotest-qemu.virtio_console.spread_2.unix_socket.virtserialport_test.restart.stressed: PASS (216.70 s)

File: shared/scripts/virtio_console_guest.py [hunk 2]
```diff
@@ -273,7 +277,10 @@ def _get_port_status(self, in_files=None):
                         port[m.group(1)] = m.group(2)
 
                     if port['is_console'] == "yes":
-                        port["path"] = "/dev/hvc%s" % (port["console_vtermno"])
+                        port_num = int(port["console_vtermno"]) - 1 if \
```
Comment (vivianQizhu) [inline shared/scripts/virtio_console_guest.py]: Better avoid use `\`, if line is too long I think we would rather use 'if ... : else: ...'

Comment (vivianQizhu) [reply shared/scripts/virtio_console_guest.py]: Or cleaner
```
port_num = int(port["console_vtermno"])
if current_version >= fixed_version:
    port_num -= 1
```

Comment (nanliu-r) [reply shared/scripts/virtio_console_guest.py]: It looks better, update and thanks.

File: shared/scripts/virtio_console_guest.py [hunk 3]
```diff
@@ -24,6 +24,8 @@
 import six
 from six.moves import xrange
 from six.moves import input
+import platform
```
Comment (lolyu) [inline shared/scripts/virtio_console_guest.py]: please keep your import statements ordered.

Comment (nanliu-r) [reply shared/scripts/virtio_console_guest.py]: Thanks and update.

**Comment by vivianQizhu** [comment]
Ack.

**Pull request merged**